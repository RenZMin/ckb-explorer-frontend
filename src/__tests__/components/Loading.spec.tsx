import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Loading from '../../components/Loading'
import SmallLoading from '../../components/Loading/SmallLoading'
import { LoadingPanel } from '../../components/Loading/styled'
import { SmallLoadingPanel } from '../../components/Loading/SmallLoading/styled'

describe('Loading Component', () => {
  let loading: ReactElement
  let smallLoading: ReactElement

  beforeAll(() => {
    loading = <Loading show />
    smallLoading = <SmallLoading />
  })

  it('shallow renders', () => {
    let wrapper = renderer.create(loading).toJSON()
    expect(wrapper).toMatchSnapshot()

    wrapper = renderer.create(smallLoading).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    let wrapper = shallow(loading)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(LoadingPanel).children().length).toBe(1)

    wrapper = shallow(smallLoading)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(SmallLoadingPanel).children().length).toBe(1)
  })
})
