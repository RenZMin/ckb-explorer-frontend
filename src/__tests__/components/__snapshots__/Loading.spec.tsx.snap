// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component shallow renders 1`] = `
<div
  className="sc-bdnxRM bYhsjP"
>
  <img
    alt="loading"
    src="loading.gif"
  />
</div>
`;

exports[`Loading Component shallow renders 2`] = `
<div
  className="sc-gtsrHT KCwSX"
>
  <img
    alt="loading"
    src="loading.gif"
  />
</div>
`;
